File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        best = 0 
        left = 0
        right = len(heights) - 1

        while left < right:
            current = min(heights[left], heights[right]) * (right - left)
            best = max(best, current)
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1 
    
        return best
            
            
        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int left = 0;
        int right = 1;
        int best = 0;

        while (right < prices.size()) {
            if (prices[left] > prices[right]) {
                left = right;
                right += 1;
            }   else {
                best = max(best,prices[right] - prices[left]);
                right += 1;
            }
        }
        return best;
    }
};
